code/modules/surgery/anesthesia_machine.dm

```diff
@@ -39,28 +39,39 @@
 /obj/machinery/anesthetic_machine/attack_hand(mob/living/user)
 	. = ..()
 	if(retract_mask())
-		visible_message("<span class='notice'>[user] retracts the mask back into the [src].</span>")
+		visible_message(span_notice("[user] retracts the mask back into the [src]."))
+	else if(attached_tank)// If attached tank, remove it.
+		attached_tank.forceMove(loc)
+		to_chat(user, span_notice("You remove the [attached_tank]."))
+		attached_tank = null
+		update_appearance(UPDATE_ICON)
+		if(mask_out)
+			retract_mask()
+
+/obj/machinery/anesthetic_machine/AltClick(mob/user)
+	. = ..()
+	if(attached_tank && mask_out)
+		to_chat(user, span_warning("Disconnect the tank from the person first!"))
+		return
+	else
```

Comment:
Maybe add a delay to changing it back to an iv drip
That way if someone accidentally alt click it they have a window to cancel it

Reply:
fair